model/src/pyrenew/latent/hospitaladmissions.py

```diff
@@ -184,13 +211,33 @@ def sample(
             mode="full",
         )[: infection_hosp_rate_t.shape[0]]
 
-        # Applying the day of the week effect
-        latent_hospital_admissions = (
-            latent_hospital_admissions
-            * self.day_of_week_effect_rv(
-                n_timepoints=latent_hospital_admissions.size, **kwargs
-            )[0].value
-        )
+        # Applying the day of the week effect. For this we need to:
+        # 1. Get the day of the week effect
+        # 2. Identify the offset of the latent_infections
+        # 3. Apply the day of the week effect to the latent_hospital_admissions
+        dow_effect_sampled = self.day_of_week_effect_rv(**kwargs)[0]
+
+        if dow_effect_sampled.value.size != 7:
+            raise ValueError(
+                "Day of the week effect should have 7 values. "
+                f"Got {dow_effect_sampled.value.size} instead."
+            )
+
+        # Identifying the offset
+        inf_offset = (
+            latent_infections.t_start % 7
+            if latent_infections.t_start is not None
+            else 0
+        ) + self.obs_data_first_day_of_the_week
+
+        # Replicating the day of the week effect to match the number of
+        # timepoints
+        dow_effect = jnp.tile(
```

Comment:
Why aren't we using `tile_unitl_n`?

Reply:
This implementation was prior to that function. Will fix.